test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
You could pull out something like a `same_value()` function so you
don't have to repeat the logic.

```
void expect_same(double x, double y) {
  if (isnan(x) && isnan(y)) return;
  EXPECT_EQ(x, y);
}
```

But I don't see how to do this test for expected NaN values any
other way.

> On Apr 28, 2016, at 3:32 PM, rayleigh <notifications@github.com> wrote:
> 
> In test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp:
> 
> >  
> >  static inline void expect_val_deriv_eq(double exp_var, double test_var) {
> > +  using stan::math::is_nan;
> > +  if (is_nan(exp_var) && is_nan(test_var)) return;
> 
> Daniel, checking that both the expected and test values are NaN isn't a suggestion. It's what I ultimately implemented in this pull request. I did so by:
> 1. Using return to exit early from functions (the code above) if both expected and test values are NaN
> 2. Using continue in a loop if both expected and test values are NaN
> 3. If the two options above are not possible, push the check within an if block
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>